tests/pos/TastyADT.scala

```diff
@@ -76,39 +71,39 @@ object tasty {
     case Ident(name: TermName, override val tpe: Type)
     case Select(prefix: Term, name: PossiblySignedName)
     case Literal(value: Constant)
-    case This(id: Id | Empty)
-    case New(tpt: Term)
+    case This(id: Option[Id])
+    case New(tpt: TypeTree)
     case NamedArg(name: TermName, arg: Term)
     case Apply(fn: Term, args: List[Term])
-    case TypeApply(fn: Term, args: List[Term])
-    case Super(thiz: Term, mixin: Id | Empty)
-    case Typed(expr: Term, tpt: Term)
+    case TypeApply(fn: Term, args: List[TypeTree])
+    case Super(thiz: Term, mixin: Option[Id])
+    case Typed(expr: Term, tpt: TypeTree)
     case Assign(lhs: Term, rhs: Term)
     case Block(stats: List[Statement], expr: Term)
     case Inlined(call: Term, bindings: List[Definition], expr: Term)
-    case Lambda(method: Term, tpt: Term | Empty)
+    case Lambda(method: Term, tpt: Option[TypeTree])
```

Comment:
Why is tpt optional here?

Reply:
It should probably not be. https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/core/tasty/TastyFormat.scala#L88

Reply:
Actually, the doc is wrong. https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala#L1022-L1025

The docs should be updated to 
```
LAMBDA         Length meth_Term target_Type?
```

Reply:
Created #4285 to fix this